examples/armv4t/gdb/host_io.rs

```diff
@@ -113,8 +127,9 @@ impl target::ext::host_io::HostIoPread for Emu {
         offset: u32,
         output: HostIoOutput<'a>,
     ) -> HostIoResult<HostIoToken<'a>, Self> {
-        if fd < FD_MAX {
-            if let Some(ref mut file) = self.files[fd as usize] {
+        let fd: usize = fd as usize;
+        if fd < self.files.len() {
+            if let Some(ref mut file) = self.files[fd] {
```

Comment:
you can collapse these nested calls to just

```rust
if let Some(file) = self.files.get_mut(fd) {

}
```

(this applies to all other instances of this pattern)